include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -210,14 +210,23 @@ namespace CUDAWrappers
 
     /**
      * This method runs the loop over all cells and apply the local operation on
-     * each element in parallel. @p func is a functor which is appplied on each color.
+     * each element in parallel. @p func is a functor which is applied on each color.
      */
     template <typename functor, typename VectorType>
     void
     cell_loop(const functor &   func,
               const VectorType &src,
               VectorType &      dst) const;
 
+    /**
+     * This method runs the loop over all cells and apply the local operation on
+     * each element in parrallel. This function is very similar to cell_loop()
+     * but it uses a simpler functor.
+     */
+    template <typename functor>
+    void
+    evaluate_coefficients(functor &func) const;
```

Comment:
Can you clarify what the expected signature for the call to `functor` object is here and in the other places where `functor` is used?

Also, for me looks a bit strange to not capitalize the first letter of `functor`. Do you have a good reason for this choice?

Reply:
> Why can't the functor be const?

The reason the functor is not `const` is because the coefficients are data of the functor and we want to be able to change them. Now it's true that the data will (always?) be a pointer so it could be `const` but I think that makes it more clear that you will change the data store in functor.

> Also, for me looks a bit strange to not capitalize the first letter of functor. Do you have a good reason for this choice?

No reason. I don't think we have rule (we use both `Number` and `number`) but I'll change it.

Reply:
I'll point out that if the object is not `const`, then you can't bind an rvalue temporary value to it. That is, you can't for example pass in a lambda function here.

Reply:
So if you don't want to make it `const`, don't make it a reference either.